bundle/deploy/terraform/plan.go

```diff
@@ -9,8 +9,54 @@ import (
 	"github.com/databricks/cli/libs/cmdio"
 	"github.com/databricks/cli/libs/terraform"
 	"github.com/hashicorp/terraform-exec/tfexec"
+	tfjson "github.com/hashicorp/terraform-json"
 )
 
+// printPlanSummary prints a high level summary of the terraform plan that will
+// be applied during bundle deploy / destroy.
+func printPlanSummary(ctx context.Context, plan *tfjson.Plan) error {
+	cmdio.LogString(ctx, "\nPlan:")
+	for _, change := range plan.ResourceChanges {
+		tfActions := change.Change.Actions
+		if tfActions.Read() || tfActions.NoOp() {
+			continue
+		}
+
+		var action string
+		switch {
+		case tfActions.Update():
+			action = "Update"
+		case tfActions.Create():
+			action = "Create"
+		case tfActions.Delete():
+			action = "Delete"
+		case tfActions.Replace():
+			action = "Replace"
+		default:
+			return fmt.Errorf("unknown terraform actions: %s", tfActions)
+		}
+
+		resourceType := change.Type
+		switch resourceType {
+		case "databricks_job":
+			resourceType = "Job"
+		case "databricks_pipeline":
+			resourceType = "DLT Pipeline"
+		case "databricks_mlflow_model":
+			resourceType = "Mlflow Model"
+		case "databricks_mlflow_experiment":
+			resourceType = "Mlflow Experiment"
+		}
```

Comment:
Could we maintain this list somewhere more central, so it's easier for maintainers who add new resource types?

Reply:
And shouldn't we just print the number of resources instead? So our log becomes shorter? It's very common for users to do frequent deploys during development. It would be good if the logs are no more than 5 lines of output.

This is what dbt's output looks like:

```
$ dbr run
Running with dbt=0.20.0
Found 4 models, 8 tests, 2 snapshots, 0 analyses, 301 macros, 0 operations, 0 seed files, 0 sources

14:30:01 | Concurrency: 1 threads (target='dev')
14:30:01 |
14:30:01 | 1 of 4 START view model my_dbt_project.my_first_model.......... [RUN]
```

for us just saying N files, N resources could be fine. and then we can print the file path after.